app/editor/src/store/slices/content/useContentStore.ts

```diff
@@ -40,8 +40,8 @@ export const useContentStore = (props?: IContentProps): [IContentState, IContent
   );
 
   useDeepCompareEffect(() => {
-    if (props?.filter) {
-      controller.storeFilter(props?.filter);
+    if (!state.initialized && props?.filter) {
```

Comment:
This fixed the issue with the users filter getting reset when returning to the page.